src/test/txreconciliation_tests.cpp

```diff
@@ -0,0 +1,686 @@
+    assert(!tracker_test1.RespondToReconciliationRequest(skdata));
+
+    // The node receives a reconciliation request and should respond with a sketch after a delay.
+    TxReconciliationTrackerTest tracker_test2(true);
+    tracker_test2.HandleReconciliationRequest();
+    // Too early, do not respond yet.
+    assert(!tracker_test2.RespondToReconciliationRequest(skdata));
+    SetMockTime(start_time + 8000);
+    assert(tracker_test2.RespondToReconciliationRequest(skdata));
+
+    // Request at the wrong reconciliation phase should be ignored.
+    SetMockTime(start_time + 12000);
+    tracker_test2.HandleReconciliationRequest();
+    assert(!tracker_test2.RespondToReconciliationRequest(skdata));
+}
+
+BOOST_AUTO_TEST_CASE(RespondToReconciliationRequestTest)
+{
+    int64_t start_time = GetTime();
+    std::vector<uint8_t> skdata;
+    SetMockTime(start_time);
+
+    // Check that we won't respond if the peer is removed ("unregistered").
+    TxReconciliationTracker tracker;
+    NodeId peer_id0 = 0;
+    tracker.SuggestReconciling(peer_id0, true);
+    assert(tracker.EnableReconciliationSupport(peer_id0, true, true, false, 1, 0));
+    tracker.HandleReconciliationRequest(peer_id0, 1, 1);
+    SetMockTime(start_time + 1000);
+    tracker.RemovePeer(peer_id0);
+    assert(!tracker.RespondToReconciliationRequest(peer_id0, skdata));
+
+    // The node receives a reconciliation request noting that the initiator has an empty set.
+    // The node should terminate reconciliation by sending an empty sketch.
+    TxReconciliationTrackerTest tracker_test(true);
+    tracker_test.HandleReconciliationRequest(1);
+    SetMockTime(start_time + 2000);
+    assert(tracker_test.RespondToReconciliationRequest(skdata));
+    assert(skdata.size() == 0);
+
+    // The node receives a reconciliation request noting that their local set is empty.
+    // The node should terminate reconciliation by sending an empty sketch.
+    TxReconciliationTrackerTest tracker_test2(true);
+    tracker_test2.AddTransactions(5);
+    tracker_test2.HandleReconciliationRequest(0);
+    SetMockTime(start_time + 3000);
+    assert(tracker_test2.RespondToReconciliationRequest(skdata));
+    assert(skdata.size() == 0);
+
+    // The node receives a reconciliation request and should respond with an expected sketch.
+    TxReconciliationTrackerTest tracker_test3(true);
+    double q = 1;
+    tracker_test3.AddTransactions(10);
+    tracker_test3.HandleReconciliationRequest(1, q * Q_PRECISION);
+    SetMockTime(start_time + 4000);
+    assert(tracker_test3.RespondToReconciliationRequest(skdata));
+    uint32_t expected_capacity = (10 - 1) + q * 1 + 1;
+    Minisketch expected_sketch = tracker_test3.ComputeSketch(tracker_test3.m_transactions, expected_capacity);
+    assert(skdata == expected_sketch.Serialize());
+    // Then respond with an extension sketch.
+    tracker_test3.HandleExtensionRequest();
+    assert(tracker_test3.RespondToReconciliationRequest(skdata));
```

Comment:
Adding a check for `local_set`, but it's hard to get into `local_set_snapshot` because it's a private variable with no getter method. I could possibly check it implicitly, but that's more of a functional test thing.